### PR TITLE
feat: Add soft-delete for incidents through the status

### LINKED ISSUE #2909: [➕ Feature]: See deleted/resolved incidents
As a user, I would like to see deleted/resolved incidents in a different panel than the default incident page, or at least filter them by default.

As a user, right now, I cannot see incidents that were deleted.

### PR DESCRIPTION
**Pull request by VladimirFilonov**
Closes #2909

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: keep/api/models/alert.py [hunk 1]
```diff
@@ -113,6 +113,8 @@ class IncidentStatus(Enum):
     ACKNOWLEDGED = "acknowledged"
     # Incident was merged with another incident
     MERGED = "merged"
+    # Incident was merged with another incident
```
Comment (talboren) [inline keep/api/models/alert.py:116]: fix comment

File: tests/test_incidents.py [hunk 2]
```diff
@@ -227,7 +227,7 @@ def test_add_remove_alert_to_incidents(db_session, setup_stress_alerts_no_elasti
 def test_get_last_incidents(db_session, create_alert):
```
Comment (talboren) [inline tests/test_incidents.py:227]: can we add a test for deleting an incident and making sure that it gets the right value?

File: keep/api/core/db.py [hunk 3]
```diff
@@ -3489,31 +3481,30 @@ def get_incident_by_fingerprint(tenant_id: str, fingerprint: str) -> Optional[In
 def delete_incident_by_id(
     tenant_id: str,
     incident_id: UUID,
+    deleted_by: str | None = None,
```
Comment (talboren) [inline keep/api/core/db.py:3484]: use deleted_by when you update?

File: keep/api/core/db.py [hunk 4]
```diff
@@ -2921,25 +2922,16 @@ def is_alert_assigned_to_incident(
     with Session(engine) as session:
         assigned = session.exec(
             select(LastAlertToIncident)
+            .join(Incident, LastAlertToIncident.incident_id == Incident.id)
```
Comment (talboren) [inline keep/api/core/db.py:2925]: is this a bug now?

**Review by talboren — CHANGES_REQUESTED** [review]
left a few minor comments

File: keep/api/core/db.py (hunk 4 above)
Comment (VladimirFilonov) [reply keep/api/core/db.py:2925]: Not actually. If incident is deleted we treat it as Alert is not assigned to the incident anymore

File: keep/api/core/db.py (hunk 3 above)
Comment (VladimirFilonov) [reply keep/api/core/db.py:3484]: It's leftover from the first try when I added new fields. Now we need to have it in Audit as status change. BTW, do we track status changes in IncidentAudit?

File: tests/test_incidents.py (hunk 2 above)
Comment (VladimirFilonov) [reply tests/test_incidents.py:227]: We have `test_incident_bl_delete_incident` where we checking the status (reversed, but still )

**Review by talboren — APPROVED** [review]
LGTM

**Pull request merged**